File: programme/area-of-rectangle/area-of-rectangle.c
/**
 * Computes the area of a rectangle
 */

#include <stdio.h>
#include <assert.h>         // for the assert function

double area_of_rectangle(double length, double breadth);
void test(void);

int main(void) {
    double area, length, breadth;

    test();

    printf("length: ");
    scanf("%lf", &length);
    printf("breadth: ");
    scanf("%lf", &breadth);
    area = area_of_rectangle(length, breadth);
    printf("area: %.1lf\n", area);

    return 0;
}

// calculates the area
double area_of_rectangle(double length, double breadth) {
    return length * breadth;
}

// checks correctness of the area calculation
void test(void) {
    assert(area_of_rectangle(2, 3) == 6);
}

File: programme/area-of-rectangle/areaofrectangle.go
// Write a programme to compute the area of a rectangle

package main

import (
	"fmt"
	"math"
)

func areaOfRectangle(length, breadth float64) float64 {
	return length * breadth
}

func roundFloat(val float64, precision uint) float64 {
	var ratio float64 = math.Pow(10, float64(precision))
	return math.Round(val*ratio) / ratio
}

func main() {
	var length, breadth, area float64

	fmt.Print("length  : ")
	fmt.Scan(&length)

	fmt.Print("breadth : ")
	fmt.Scan(&breadth)

	area = areaOfRectangle(length, breadth)
	fmt.Println("\narea : ", roundFloat(area, 2))
}
